aiohttp/payload.py

```diff
@@ -440,6 +440,15 @@ def __init__(self, value: StreamReader, *args: Any, **kwargs: Any) -> None:
         super().__init__(value.iter_any(), *args, **kwargs)
 
 
+class TempFileMetaclass(ABCMeta):
```

Comment:
I think `Protocol` and `runtime_checkable` from `typing_extensions` can serve this duty better.
See https://docs.python.org/3/library/typing.html#typing.runtime_checkable for docs.

Reply:
`Protocol` and `runtime_checkable` introduced in python 3.8, but what should we do for 3.5-3.7 then?